azure/iac/sql_servers_auditing.rego

```diff
@@ -0,0 +1,55 @@
+package rule
+
+# https://docs.microsoft.com/en-us/azure/templates/microsoft.sql/2021-02-01-preview/servers/auditingsettings
+
+#
+# PR-AZR-0134-ARM
+#
+
+default sql_server_log_audit = null
+
+azure_attribute_absence["sql_server_log_audit"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.sql/servers/auditingsettings"
```

Comment:
`microsoft.sql/servers/auditingsettings` is a sub resource of `microsoft.sql/servers` . first you need to check the existence of this resource, then checking if it is enabled. it is possible that this resource is not available and we are skipping the test: https://github.com/Azure/azure-quickstart-templates/blob/master/quickstarts/microsoft.sql/sql-auditing-server-policy-to-blob-storage/azuredeploy.json

Reply:
@rezoan your thought?

Reply:
@farchide `auditingSettings `can be added to the ARM template in 2 way, basically most of the dependent resource can be set in 2 way:
- one way to add the resource inside child `resources `array, like you have shown at https://github.com/Azure/azure-quickstart-templates/blob/master/quickstarts/microsoft.sql/sql-auditing-server-policy-to-blob-storage/azuredeploy.json
In this case we can set the type as `auditingsettings`. fully qualified type is not necessary and `dependsOn ` array property also not necessary but if we add wont hurt and its a good practice.
- Another way is to add as sibling resource in parallel like @shahinshirvani checked in his rule. 
While adding the resource as sibling, we need to set the fully qualified type as `"type": "microsoft.sql/servers/auditingsettings" ` and `dependsOn` array property is mandatory to indicate on which sibling resource it depends on. could be multiple resource. Also in this case make sure `dependsOn` has fully qualified server name . e.g. `Microsoft.Sql/servers/{server_name_from_arm_parameter}`



**My suggestion here would be to create 2 separate rule to cover both scenario for such resources.**
**Can follow `sql_alert_policy.rego`, it has the same scenario already handled.**

Reply:
I am of same mind with @rezoan  on this
so I updated the rego file